pkg/volume/plugins.go

```diff
@@ -641,33 +641,35 @@ func (pm *VolumePluginMgr) initProbedPlugin(probedPlugin VolumePlugin) error {
 // support it, return error.
 func (pm *VolumePluginMgr) FindPluginBySpec(spec *Spec) (VolumePlugin, error) {
 	pm.mutex.Lock()
-	defer pm.mutex.Unlock()
 
 	if spec == nil {
+		pm.mutex.Unlock()
 		return nil, fmt.Errorf("Could not find plugin because volume spec is nil")
 	}
 
-	matchedPluginNames := []string{}
 	matches := []VolumePlugin{}
-	for k, v := range pm.plugins {
+	for _, v := range pm.plugins {
 		if v.CanSupport(spec) {
-			matchedPluginNames = append(matchedPluginNames, k)
 			matches = append(matches, v)
 		}
 	}
 
 	pm.refreshProbedPlugins()
 	for _, plugin := range pm.probedPlugins {
 		if plugin.CanSupport(spec) {
-			matchedPluginNames = append(matchedPluginNames, plugin.GetPluginName())
 			matches = append(matches, plugin)
 		}
 	}
+	pm.mutex.Unlock()
```

Comment:
Generally this is considered unacceptable in our style.  A panic in CanSupport would leave the plugin wedged.

If you want to do this, reorganize into two sets of code and continue to use defer

/hold

Reply:
@smarterclayton 
I have restored the lock deferring.